src/main/java/net/minecraftforge/coremod/api/ASMAPI.java

```diff
@@ -70,39 +116,107 @@ public static boolean getSystemPropertyFlag(final String propertyName) {
         return Boolean.getBoolean(propertyName) || Boolean.getBoolean("coremod." + propertyName);
     }
 
+    /**
+     * The mode in which the given code should be inserted.
+     */
     public enum InsertMode {
         REMOVE_ORIGINAL, INSERT_BEFORE, INSERT_AFTER
     }
 
     /**
-     * Finds the first instruction with matching opcode
+     * The type of instruction. Useful for searching for a specfic instruction, and is preferred over checking the
+     * opcode or other equivalent.
+     *
+     * @see AbstractInsnNode
+     */
+    public enum InsnType {
+        INSN,
```

Comment:
This has the ordinal compared to the ASM type number. Is there a way we can just expose the ASM type as API instead so we don't have to keep this enum in sync with ASM?

Reply:
Right, so these types are merely mimicking the constants in `AbstractInsnNode`. This enum was my simplest implementation of it. It is possible to directly invoke them by importing that class in the JS file, but that's what I'm trying to avoid...

Reply:
Humm, it would make the code more verbose, but less error prone to instead just copy these constants here instead of using a enum and hoping the ordinals align.
Not sure what the best option would be here. This works fine. But I just dont like the fragility of it.

Reply:
I am not opposed to verbosity, but I don't want to bloat it either. Given that it would only benefit us to directly link them, I might as well just do that.

Reply:
This is a much better/easier to maintain approach, with minimal code uglyness, i like it.